hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/qjournal/client/QuorumJournalManager.java

```diff
@@ -585,8 +585,10 @@ private void selectRpcInputStreams(Collection<EditLogInputStream> streams,
     int maxAllowedTxns = !onlyDurableTxns ? highestTxnCount :
         responseCounts.get(responseCounts.size() - loggers.getMajoritySize());
     if (maxAllowedTxns == 0) {
-      LOG.debug("No new edits available in logs; requested starting from " +
-          "ID " + fromTxnId);
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("No new edits available in logs; requested starting from " +
+            "ID " + fromTxnId);
+      }
```

Comment:
If would be better to remove the `if (LOG.isDebugEnabled())` and change the log statement to:

```
LOG.debug("No new edits available in logs; requested starting from ID {}", fromTxnId);
```

This avoids any evaluation and inside the debug method, it checks if debug is enabled or not before evaluating the string.

Reply:
Thanks for review @sodonnel ,I will change it.

Reply:
> If would be better to remove the `if (LOG.isDebugEnabled())` and change the log statement to:
> 
> ```
> LOG.debug("No new edits available in logs; requested starting from ID {}", fromTxnId);
> ```
> 
> This avoids any evaluation and inside the debug method, it checks if debug is enabled or not before evaluating the string.

This is a good solution!